### PR TITLE
Library updates and AndroidX Preferences

### PR DESCRIPTION
**Pull request by TacoTheDank**
### 1st commit
- Update .gitignore (properly exclude .idea files)
- Delete unneeded module .gitignore
- Delete nonfunctional androidTest

### 2nd commit
- Update gradle wrapper to 6.7.1 using git bash

### 3rd commit
- Delete old findbugs stuff (is outdated anyway, spotbugs should be used instead)
- Enable minify (this should be perfectly fine if -dontobfuscate is in proguard rules)
- Organize libraries
- Update Android Studio 3.2.0 -> 4.1.1

### 4th commit
- Remove unused legacy library
- Update ConstraintLayout 1.1.3 -> 2.0.4 ([changelog]())
- Update zxing-android-embedded 4.0.2 -> 4.1.0 ([changelog](https://github.com/journeyapps/zxing-android-embedded/blob/master/CHANGES.md))
- Update zxing-core 3.3.0 -> 3.4.1 ([changelog](https://github.com/zxing/zxing/releases))
- Update commons-csv 1.5 -> 1.8 ([changelog](https://commons.apache.org/proper/commons-csv/changes-report.html))
- Update jaredrummler-colorpicker 1.0.2 -> 1.1.0 ([changelog](https://github.com/jaredrummler/ColorPicker/commits/master))
- Update Guava 20.0 -> 30.1 ([changelog](https://github.com/google/guava/releases))
- Update JUnit 4.12 -> 4.13.1 ([changelog](https://github.com/junit-team/junit4/releases))

### 5th commit
- Migrate preferences to AndroidX
- Use AndroidX Preference supported NumberPickerPreference library
- Replace CheckboxPreference with SwitchPreferenceCompat (see [this](https://developer.android.com/guide/topics/ui/settings/components-and-attributes#individual_preferences))

### 6th commit
- Resolve app name in build.gradle

I built an APK and tested. Everything worked fine.

### CODE REVIEW

**Comment by TheLastProject** [comment]
This will probably take me some time to review but thank you!

File: app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java [hunk 1]
```diff
@@ -446,7 +446,7 @@ public void onDialogDismissed(int dialogId)
                     // Nothing to do, no change made
                 }
             });
-            dialog.show(getFragmentManager(), "color-picker-dialog");
+            dialog.show(getSupportFragmentManager(), "color-picker-dialog");
```
Comment (TheLastProject) [inline app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java:449]: Hmm, Android Studio insists on telling me:

```
Cannot resolve method 'show(androidx.fragment.app.FragmentManager, java.lang.String)'
```

Yet it doesn't crash. Weeeeeird.

**Review by TheLastProject — COMMENTED** [review]
App indeed seems to run just fine still. Thank you.

I'm a bit worried about the "OutOfMemory" unit test failures in CI/CD though, but they don't happen on my own system...

File: app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java (hunk 1 above)
Comment (TheLastProject) [reply app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java:449]: Nevermind it stopped saying this. Guess it was some broken cache stuff lol.

**Comment by TheLastProject** [comment]
Oh yeah, I forgot to say: thank you for the different debug name in particular. That is an amazing improvement for my ability to both use the app and work on it.

File: app/build.gradle [hunk 2]
```diff
@@ -1,80 +1,68 @@
 apply plugin: 'com.android.application'
-apply plugin: 'findbugs'
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/build.gradle:2]: @TheLastProject I'm not well versed in it myself, either. I think the repos would be able to best explain:
- https://github.com/spotbugs/spotbugs-gradle-plugin
- https://github.com/spotbugs/spotbugs

[the remark this answers is not in the document]
Comment (TheLastProject) [reply app/build.gradle:2]: Well, findbugs has definitely been helpful, so I don't want to get rid of it until we have a working replacement.

[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/build.gradle]: Fair enough 👍

**Comment by TacoTheDank** [comment]
@TheLastProject The reason I had deleted it all is because, with Gradle 6 and up, findbugs no longer exists, so attempting to sync gradle will result in an error with the findbugs plugin being unresolved.

I have pushed a commit with a rudimentary migration of findbugs to spotbugs. It has some kinks to work out, and isn't very stable at the moment, but it generates reports for me :Y

This implementation has spotted 6 violations.

**Comment by TheLastProject** [comment]
I guess I'll just have to... make the import/export less memory-intensive. It may cause issues on some devices too so... important to improve anyway

**Comment by TheLastProject** [comment]
Hah! The OutOfMemory no longer happens! I kinda knew there were db connection leaks but I didn't actually realize they affected memory usage that much. Good thing it broke like this because I underestimated the seriousness of that issue and also to figure out where it was going wrong I needed to get the CI logs which I now do :)

**Comment by TheLastProject** [comment]
Wait... they're failing again? But they passed on https://github.com/TheLastProject/Catima/runs/1640886008?check_suite_focus=true#step:7:11! WHAT?!

**Comment by TacoTheDank** [comment]
Hoo boy...

**Comment by TacoTheDank** [comment]
Thanks for merging! Sorry about all that trouble lol

**Comment by TheLastProject** [comment]
I'm honestly kinda happy it broke like this because it forced me to tackle some technical debt :P

**Pull request merged**